docs/asyncapi.yaml

```diff
@@ -52,156 +51,148 @@ components:
-        sbi:
-          $ref: '#/components/schemas/SBI'
-        frn:
-          $ref: '#/components/schemas/FRN'
         email:
           $ref: '#/components/schemas/Email'
+        frn:
+          $ref: '#/components/schemas/FRN'
+        sbi:
+          $ref: '#/components/schemas/SBI'
         address:
           $ref: '#/components/schemas/Address'
-        scheme:
-          $ref: '#/components/schemas/Scheme'
-        payments:
+        funding:
           type: array
           items:
-            $ref: '#/components/schemas/Payment'
-        funding:
+            - $ref: '#/components/schemas/Funding'
+            - $ref: '#/components/schemas/FundingTotal'
+        payments:
           type: array
           items:
-            $ref: '#/components/schemas/Funding'
+            $ref: '#/components/schemas/Payment'
+        scheme:
+          $ref: '#/components/schemas/Scheme'
     BusinessName:
       type: string
       description: The business name
-      example: 'FFC Ltd'
-    SBI:
-      type: number
-      description: Single Business Identifier
-      minimum: 105000000
-      maximum: 999999999
-      example: 123456789
+      example: 'Summer Days Farm'
+    Email:
+      type: string
+      description: The business email address
+      example: farm@farms.com
     FRN:
       type: number
       description: Firm Reference Number
       minimum: 1000000000
       maximum: 9999999999
       example: 1234567890
-    Email:
-      type: string
-      description: The business email address
-      example: farm@farms.com
+    SBI:
+      type: number
+      description: Single Business Identifier
+      minimum: 105000000
+      maximum: 999999999
+      example: 123456789
     Address:
       type: object
       properties:
         line1:
           type: string
           description: The first line of the address
-          example: 1 The Street
+          example: 1 Some Street
         line2:
           type: string
           description: The second line of the address
-          example: The Area
+          example: Some Area
         line3:
           type: string
           description: The third line of the address
-          example: The Town
+          example: Some Town
         line4:
           type: string
           description: The fourth line of the address
-          example: The Region
+          example: Some Region
         line5:
           type: string
           description: The fifth line of the address
-          example: The County
+          example: Some County
         postcode:
           type: string
           description: The postcode
-          example: AB1 2CD
-    Scheme:
+          example: FA1 2PY
+    Funding:
       type: object
       required:
       - name
-      - shortName
-      - year
-      - frequency
-      - agreementNumber
+      - area
+      - annualValue
+      - quarterlyValue
+      - quarterlyReduction
+      - quarterlyPayment
       properties:
         name:
```

Comment:
These should all be schemas and referenced

Reply:
What do you mean exactly? Wouldn't referencing 'Funding' be reference enough?

Reply:
So this `Funding` object has got a number of properties

If we convert all of those properties into schemas, i.e. name, level, area, etc. then reference them inside this `Funding` object schema.

There are a number of benefits:

1. We can easily see what fields/objects exist within any potential message -- rather than them being hidden inside another object like here
2. We can reuse them inside other objects, e.g. `Funding` and `FundingTotal` share a lot of same properties
3. Provides a single source of truth: define once, use lots